server/src/routes/api/v1/auth/index.ts

```diff
@@ -0,0 +1,41 @@
+import { Router, Request, Response, NextFunction } from "express";
+import { AuthService } from "./../../../../logic/services/auth.service";
+import { ErrorUtils } from "./../../../../logic/utils/errorUtils";
+import { ApiError, DbError, SocoboUser } from "./../../../../models/index";
+
+
+export class AuthRouteV1 {
+  constructor (private _authService: AuthService, private _router: Router) {}
+
+  createRoutes (): Router {
+    // login the user
+    this._router.post("/login", (req: Request, res: Response, next: NextFunction) => {
+      this._authService.login(req.body.email, req.body.password)
+        .then((result: SocoboUser) => res.status(200).json(result))
+        .catch((error: any) => {
+          if (ErrorUtils.notFound(error)) {
+            res.status(404).json(
+                new DbError(`The user login is failed - ${error.message}!`, AuthService.name, 
+                              "login(email,password)", error).forResponse());
+          } else {
+            res.status(500).json(
+                new ApiError(`Internal Server Error: ${error.message}`, AuthService.name, 
+                              "login(email,password)", error).forResponse());
+          }
+        });
+    });
+
+    // register the user
+    this._router.post("/register", (req: Request, res: Response, next: NextFunction) => {
+      this._authService.register(req.body.email, req.body.password)
+        .then((result: SocoboUser) => res.status(200).json(result))
```

Comment:
I would never return the password regardless of whether it is a hash. The client doesn't need this information. 

And I would maybe add general information about the user (no sensitive and only things like isAdmin and so) to the payload of the jwt since the jwt will be stored into a cookie, the client has always access to these information via the jwt and does not request these again and again